src/scripts/create_hunter_csv.py

```diff
@@ -0,0 +1,331 @@
+        path_to_each_test_json = get_paths_to_json(path_w_date)
+
+        list_of_hunter_df = []
+        list_of_type_of_tests = []
+        for test_json_path in path_to_each_test_json:
+            hunter_df, type_of_test = get_hunter_df_w_test_type(test_json_path)
+            if type_of_test != '' and not hunter_df.empty:
+                list_of_hunter_df.append(hunter_df)
+                list_of_type_of_tests.append(type_of_test)
+
+        # Get concatenated hunter dfs across all test types currently supported
+        concat_hunter_data_frames = {
+            SUBSTR_TESTS_NAMES[0]: hunter_df_fixed_100,
+            SUBSTR_TESTS_NAMES[1]: hunter_df_rated_100,
+            SUBSTR_TESTS_NAMES[2]: hunter_df_fixed_1000,
+            SUBSTR_TESTS_NAMES[3]: hunter_df_rated_1000,
+            SUBSTR_TESTS_NAMES[4]: hunter_df_fixed_10000,
+            SUBSTR_TESTS_NAMES[5]: hunter_df_rated_10000
+        }
+
+        for i, test_type in enumerate(list_of_type_of_tests):
+            for test_partition, hunter_df in concat_hunter_data_frames.items():
+                if test_partition in test_type:
+                    concat_df = pd.concat([hunter_df, list_of_hunter_df[i]])
+                    concat_hunter_data_frames[test_partition] = concat_df
+                    break
+                else:
+                    get_error_log(test_type)
+
+        # Save two versions of the df: 1) with the Cassandra git shas only (for hunter), 2) with two git shas (of the
+        # Cassandra and fallout-tests repos) for auditability
+        for i, test_name in enumerate(LWT_TESTS_NAMES):
+            for substr_test_name in SUBSTR_TESTS_NAMES:
+                if substr_test_name in test_name:
+                    df_w_two_git_sha = concat_hunter_data_frames[substr_test_name]
+                    save_df_to_csv(
+                        df_w_two_git_sha,
+                        f'{HUNTER_CSV_PROJ_DIR}{os.sep}{HUNTER_PREFIX}{test_name}{TWO_GIT_SHA_SUFFIX}{HUNTER_FILE_FMT}'
+                    )
+                    df_w_one_git_sha = df_w_two_git_sha.drop(
+                        FALLOUT_TESTS_COL_NAME, axis=1)
+                    save_df_to_csv(
+                        df_w_one_git_sha,
+                        f'{HUNTER_CSV_PROJ_DIR}{os.sep}{HUNTER_PREFIX}{test_name}{HUNTER_FILE_FMT}'
+                    )
+
+    else:
+        hunter_df_100_fixed, hunter_df_100_rated, hunter_df_1000_fixed, hunter_df_1000_rated, hunter_df_10000_fixed, \
+            hunter_df_10000_rated, types_of_tests = [], [], [], [], [], [], []
+        for input_date in nightly_result_dates:
+            for test_json_path in get_paths_to_json(f'{NIGHTLY_RESULTS_DIR}{os.sep}{input_date}'):
+                hunter_df, type_of_test = get_hunter_df_w_test_type(
+                    test_json_path)
+                if type_of_test:
+                    types_of_tests.append(type_of_test)
+                if SUBSTR_TESTS_NAMES[0] in type_of_test and not hunter_df.empty:
```

Comment:
I like this change, it's almost what I expected but can't you iterate over the SUBSTR_TESTS_NAMES and collapse all these if/elif into one now we have this dict? e.g.

```
for t in SUBSTR_TESTS_NAME:
   if t in type_of_test and not hunter_df.empty:
       SUBSTR_TESTS_NAME[t].append(hunter_df)
    else:
        get_error_log(type_of_test)
```

?

Reply:
I tried running the code above but I got: 
`File "create_hunter_csv.py", line 287, in <module>
    SUBSTR_TESTS_NAMES[t].append(hunter_df)
TypeError: list indices must be integers or slices, not str`. 

And also, tried:
`for i, t in enumerate(SUBSTR_TESTS_NAMES):
                    if t in type_of_test and not hunter_df.empty:
                        SUBSTR_TESTS_NAMES[i].append(hunter_df)
                    else:
                        get_error_log(type_of_test)`

But still erroring. As you have said we can just create a follow up GH issue to record these two points and address it later.